core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -159,6 +159,23 @@ public void log(String format, Object... args) {
         }
     }
 
+    /**
+     * This will log {@link RuntimeException}, if logging is enabled , and throw the runtime exception.
+     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
+     * @throws RuntimeException which is requested by this call.
+     */
+    public void logAndThrow(RuntimeException runtimeException) {
+        if (runtimeException == null) {
+            return;
+        }
+
+        // it will only log if log level is enabled in configuration
+        if (canLogAtLevel(level)) {
+            logger.error("", runtimeException);
```

Comment:
Can we use `runtimeException.getMessage()` instead of empty string?

Reply:
This is needed as the exception will be redacted from logging if the logging level doesn't include debugging/verbose logs.

Reply:
In case where client has not enabled logging. We honour that and do not expose anything in logs. We show more bias towads not looging PII.